### PR TITLE
Refactor and enhance QRM plugins support for reclaim_cores with numa_binding

### PR DESCRIPTION
**Pull request by luomingmeng**
#### What type of PR is this?

Enhancements
#### What this PR does / why we need it:
This MR introduces the following main changes to support reclaim_cores with numa_binding:
- Extract common fields from AllocationInfo into metadata and util #702
- Update plugin state via LW interface
- Adapt CPU plugin for RNB admission and provision
- Adapt memory plugin for RNB admission and provision
- Support CPU suppression and eviction for RNB
- Support CPU idle for RNB
- Update CPU plugin for pod NUMA annotations
- Enable CPU plugin to set cgroup path via Allocate Response annotations

### CODE REVIEW

File: cmd/katalyst-agent/app/options/qrm/memory_plugin.go [hunk 1]
```diff
@@ -81,6 +84,7 @@ func NewMemoryOptions() *MemoryOptions {
 		EnableMemoryAdvisor:        false,
 		EnableOOMPriority:          false,
 		EnableNonBindingShareCoresMemoryResourceCheck: true,
+		NUMABindResultResourceAllocationAnnotationKey: consts.QRMResourceAnnotationKeyNUMABindResult,
```
Comment (xu282934741) [inline cmd/katalyst-agent/app/options/qrm/memory_plugin.go:87]: The variable `EnableNUMAAllocationReactor` is also explicitly set to `false`,Similarly, all newly added flags should explicitly be assigned default values.

File: pkg/agent/qrm-plugins/advisorsvc/advisor_svc.pb.go [hunk 2]
```diff
@@ -2033,7 +2033,7 @@ func (m *ContainerMetadata) Unmarshal(dAtA []byte) error {
 					if err != nil {
 						return err
 					}
-					if skippy < 0 {
```
Comment (xu282934741) [inline pkg/agent/qrm-plugins/advisorsvc/advisor_svc.pb.go:2036]: Why would this file change？

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go [hunk 3]
```diff
@@ -100,22 +101,28 @@ type DynamicPolicy struct {
...
+	qosConfig                                 *generic.QoSConfiguration
+	dynamicConfig                             *dynamicconfig.DynamicAgentConfiguration
+	podDebugAnnoKeys                          []string
+	podAnnotationKeptKeys                     []string
+	podLabelKeptKeys                          []string
+	transitionPeriod                          time.Duration
+	cpuNUMAHintPreferPolicy                   string
+	cpuNUMAHintPreferLowThreshold             float64
+
+	reservedReclaimedCPUsSize                 int
```
Comment (xu282934741) [inline pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy.go:123]: Why does the implementation of RNB require modifications to the reservedReclaim logic?

File: pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_advisor_handler.go [hunk 4]
```diff
@@ -411,12 +444,17 @@ func (p *DynamicPolicy) handleAdvisorCPUSetMems(
...
 
-	numaSetChangedContainers := make(map[string]map[string]*state.AllocationInfo)
-	p.updateNUMASetChangedContainers(numaSetChangedContainers, allocationInfo, cpusetMems)
-	err = p.migratePagesForNUMASetChangedContainers(numaSetChangedContainers)
-	if err != nil {
-		return fmt.Errorf("migratePagesForNUMASetChangedContainers failed with error: %v", err)
-	}
+	general.Infof("dry-run set cpuset.mems for container: %s/%s from %s to: %s",
+		allocationInfo.PodNamespace, allocationInfo.PodName, allocationInfo.NumaAllocationResult.String(), cpusetMems.String())
+	// allocationInfo.NumaAllocationResult = cpusetMems
```
Comment (xu282934741) [inline pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_advisor_handler.go:449]: What does this comment mean, and is there a need to add a comment to this section?

File: cmd/katalyst-agent/app/options/qrm/cpu_plugin.go [hunk 5]
```diff
@@ -54,11 +56,12 @@ func NewCPUOptions() *CPUOptions {
...
-			EnableCPUPressureEviction: false,
-			EnableSyncingCPUIdle:      false,
-			EnableCPUIdle:             false,
-			CPUNUMAHintPreferPolicy:   cpuconsts.CPUNUMAHintPreferPolicySpreading,
+			EnableCPUAdvisor:                 false,
+			EnableCPUPressureEviction:        false,
+			EnableSyncingCPUIdle:             false,
+			EnableCPUIdle:                    false,
+			CPUNUMAHintPreferPolicy:          cpuconsts.CPUNUMAHintPreferPolicySpreading,
+			CPUNUMAReclaimedHintPreferPolicy: cpuconsts.CPUNUMAHintPreferPolicyNone,
```
[the remark this answers is not in the document]
Comment (luomingmeng) [reply cmd/katalyst-agent/app/options/qrm/cpu_plugin.go]: there are unused parameters, I deleted them

**Pull request merged**